src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let result = [];
+
+  for (const action of actions) {
+    const newState = structuredClone(state);
+
+    switch (action.type) {
```

Comment:
Use destruction for `action`
```suggestion
  for (const action of actions) {
    const newState = structuredClone(state);
    const { type, .......... } = action;

    switch (type) {
```